src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -107,14 +107,14 @@ julia> permutation_matrix(QQ,s)
 
 ```
 """
-function permutation_matrix(F::Ring, Q::AbstractVector{<:IntegerUnion})
+function permutation_matrix(R::NCRing, Q::AbstractVector{<:IntegerUnion})
    @assert Set(Q)==Set(1:length(Q)) "Invalid input"
-   Z = zero_matrix(F,length(Q),length(Q))
+   Z = zero_matrix(R,length(Q),length(Q))
    for i in 1:length(Q) Z[i,Q[i]] = 1 end
```

Comment:
```suggestion
   for i in 1:length(Q) Z[i,Q[i]] = one(R) end
```
This should be slightly more efficient for most rings, as it does not need to go through general integer coercion

Reply:
And please add some new lines to the loop, to help with readability

Reply:
Since @joschmitt is just changing the signature, this PR is fine as it is now?

Reply:
So I did some basic formatting now. From what I see, changing `1` to `one(R)` actually increases the number of allocations if `R == QQ` (which makes sense to me because then we don't allocate for the fmpq just to put it in the  matrix). Anyway, this seems to be a performance discussion beyond the scope of this pull request. I changed it to `one(R)` now because it increases readability (for me).